### ISSUE TITLE
http2 mode not working

### ISSUE THREAD

**Issue by elvizlai** [bungle/lua-resty-template#44]
for http2 request, it's not working.

for example:
```
curl --http1.1 https://example.com # works
curl https://example.com # default http2 not working
```

**Comment by bungle** [comment]
what kind of error you get? can you provide a way to reproduce?

**Comment by RandomCivil** [comment]
2021/09/06 00:48:12 [error] 15#0: *3 lua entry thread aborted: runtime error: /usr/local/openresty/site/lualib/resty/template.lua:162: http2 requests not supported yet
stack traceback:
coroutine 0:
	[C]: in function 'capture'
	/usr/local/openresty/site/lualib/resty/template.lua:162: in function 'load'
	/usr/local/openresty/site/lualib/resty/template.lua:405: in function 'parse'
	/usr/local/openresty/site/lualib/resty/template.lua:389: in function 'compile'
	/usr/local/openresty/site/lualib/resty/template.lua:637: in function 'process'
	/usr/local/openresty/site/lualib/resty/template.lua:652: in function 'render'

**Comment by RandomCivil** [comment]
l think it's because we use `location` param

```lua
local template = require "resty.template".new({
  root     = "/root/blog",
  location = "/root/blog",
})
```

just use `root` param is ok,it's work

**Comment by bungle** [comment]
@RandomCivil, yes, that is expected, `ngx.location.capture` does not work with `http2`.

**Comment by alexandrim0** [comment]
Hello!
I faced the same error.
I can`t get worked app with any configuration of package with http2.
It works with http 1.1 only.
Setting of `root` does not fix.

openresty/1.25.3.1

**Comment by alexandrim0** [comment]
It works when template is a string inside lua script and does not works when needs to read file.
